pkg/transport/generic_transport_controller.go

```diff
@@ -410,44 +465,174 @@ func (c *genericTransportController) getObjectsFromWDS(ctx context.Context, bind
+		}
+		// wrapped object name is (PlacementDecision.GetName()-WdsName).
+		// pay attention - we cannot use the PlacementDecision object name, cause we might have duplicate names coming from different WDS spaces.
+		// we add WdsName to the object name to assure name uniqueness,
+		// in order to easily get the origin PlacementDecision object name and wds, we add it as an annotations.
+		wrappedObject.SetName(fmt.Sprintf("%s-%s", binding.GetName(), c.wdsName))
+		setLabel(wrappedObject, originOwnerReferenceLabel, binding.GetName())
+		setLabel(wrappedObject, originWdsLabel, c.wdsName)
+		setAnnotation(wrappedObject, originOwnerGenerationAnnotation, binding.GetGeneration())
+		return wrappedObject, nil
+	}
+	if destToCustomizedObjects != nil {
+		asMap := map[v1alpha1.Destination]*unstructured.Unstructured{}
+		for dest, objects := range destToCustomizedObjects {
+			wrappedObject, err := wrap(objects)
+			if err != nil {
+				return nil, nil, err
+			}
+			asMap[dest] = wrappedObject
+		}
+		destToWrappedObject = util.PrimitiveMapGet(asMap)
+	} else {
+		wrappedObject, err := wrap(objectsToPropagate)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to convert wrapped object to unstructured - %w", err)
+		}
+		destToWrappedObject = func(v1alpha1.Destination) (*unstructured.Unstructured, bool) { return wrappedObject, true }
+	}
+
+	return destToWrappedObject, bindingErrors, nil
+}
+
+func (c *genericTransportController) addBindingCare(bindingName string, dest v1alpha1.Destination) {
+	c.Lock()
+	defer c.Unlock()
+	cares := c.bindingCares[bindingName]
+	if cares == nil {
+		cares = sets.New[v1alpha1.Destination](dest)
+		c.bindingCares[bindingName] = cares
+	} else {
+		cares.Insert(dest)
 	}
-	// wrapped object name is (Binding.GetName()-WdsName).
-	// pay attention - we cannot use the Binding object name, cause we might have duplicate names coming from different WDS spaces.
-	// we add WdsName to the object name to assure name uniqueness,
-	// in order to easily get the origin Binding object name and wds, we add it as an annotations.
-	wrappedObject.SetName(fmt.Sprintf("%s-%s", binding.GetName(), c.wdsName))
-	setLabel(wrappedObject, originOwnerReferenceLabel, binding.GetName())
-	setLabel(wrappedObject, originWdsLabel, c.wdsName)
-	setAnnotation(wrappedObject, originOwnerGenerationAnnotation, binding.GetGeneration())
+}
 
-	return wrappedObject, nil
+func (c *genericTransportController) setBindingCares(bindingName string, cares sets.Set[v1alpha1.Destination]) {
+	c.Lock()
+	defer c.Unlock()
+	if cares == nil {
+		delete(c.bindingCares, bindingName)
+	} else {
+		c.bindingCares[bindingName] = cares
+	}
 }
 
-func (c *genericTransportController) propagateWrappedObjectToClusters(ctx context.Context, desiredWrappedObject *unstructured.Unstructured,
+func (c *genericTransportController) enqueueBindingsForCluster(clusterAny any) {
+	cluster := clusterAny.(*clusterapi.ManagedCluster)
```

Comment:
this is coming back to using MC to store customization parameters comment. 
so now we need to cast to OCM MC. that's a bad coupling :(

Reply:
Revised to not be specific to MangedCluster.